File: core/BinaryTreeRightSideView/BinaryTreeRightSideView.java
package com.dbc;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class BinaryTreeRightSideView {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode() {
        }

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public List<Integer> rightSideView(TreeNode root) {
        if (root == null) return new ArrayList<>();
        Queue<TreeNode> queue = new LinkedList<TreeNode>(){{
            add(root);
        }};
        List<Integer> res = new ArrayList<>();
        while (!queue.isEmpty()) {
            int length = queue.size(), levelVal = 0;
            for (int i = 0; i < length; i++) {
                TreeNode node = queue.poll();
                if (node.left != null) queue.offer(node.left);
                if (node.right != null) queue.offer(node.right);
                levelVal = node.val;
            }
            res.add(levelVal);
        }
        return res;
    }
}

File: core/BinaryTreeRightSideView/BinaryTreeRightSideView.py
from typing import List
from collections import deque

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def rightSideView(root: TreeNode) -> List[int]:
    if not root:
        return []
    res, queue = [], deque([root])
    while queue:
        length = len(queue)
        level_val = 0
        for _ in range(length):
            node = queue.pop()
            if node.left:
                queue.appendleft(node.left)
            if node.right:
                queue.appendleft(node.right)
            level_val = node.val
        res.append(level_val)

    return res



